packages/components/file-card/src/file-card.js

```diff
@@ -0,0 +1,357 @@
+import { TSElement, unsafeCSS, html } from '@tradeshift/elements';
+import '@tradeshift/elements.typography';
+import '@tradeshift/elements.card';
+import '@tradeshift/elements.progress-bar';
+import '@tradeshift/elements.file-size';
+import css from './file-card.css';
+
+customElements.define(
+	'ts-file-card',
+	class extends TSElement {
+		constructor() {
+			super();
+			this.state = 'upload';
+			this.size = 'full';
+			this.removable = false;
+			this.toggleRemoveActionHoverState = this.toggleRemoveActionHoverState.bind(
+				this
+			);
+			this.toggleDownloadActionHoverState = this.toggleDownloadActionHoverState.bind(
+				this
+			);
+			this.removeFile = this.removeFile.bind(this);
+			this.downloadFile = this.downloadFile.bind(this);
+		}
+
+		static get styles() {
+			return [TSElement.styles, unsafeCSS(css)];
+		}
+
+		static get properties() {
+			return {
+				state: { type: String, reflect: true },
+				fileObject: { type: Object, attribute: 'file-object' },
+				rtl: { type: Boolean, reflect: true },
+				removable: { type: Boolean, reflect: true },
+				size: { type: String, reflect: true },
+				errorMessage: {
+					type: String,
+					reflect: true,
+					attribute: 'error-message'
+				}
+			};
+		}
+
+		get cardType() {
+			const cardTypes = {
+				uploading: 'focus',
+				download: 'default',
+				failed: 'failed'
+			};
+			return cardTypes[this.state];
+		}
+
+		get errorMessages() {
+			if (this.state === 'failed') {
+				const message = this.errorMessage;
+				return html`
+					<ts-typography
+						class="error-message"
+						text="${message}"
+						color="error"
+						variant="subtitle"
+						no-wrap
+					></ts-typography>
+				`;
+			}
+		}
+
+		get fileIcon() {
+			return html`
+				<span class="file-icon-wrapper icon-wrapper">
+					<!--TODO: use icon components-->
+					<svg
```

Comment:
Can you put he svg in a file.

Reply:
I'll do it in the next iteration, I didn't want to make a crappy icon component without enough thought in a rush👍